services/dy-static-file-server/docker/custom/Dockerfile

```diff
@@ -7,31 +7,41 @@ FROM nginx:1.21.0-alpine as base
 #     docker run dy-static-file-server:prod
 #
 
+ARG PYTHON_VERSION="3.8.10-r0"
+ARG INPUT_DIR="/www/inputs"
+ARG OUTPUT_DIR="/www/outputs"
+ARG WORKDIR="/workdir"
+
 LABEL maintainer=sanderegg
 COPY docker/custom/nginx.conf /etc/nginx/nginx.conf
 COPY docker/custom/boot.sh /boot.sh
 
-RUN apk add --update --no-cache "python3=3.8.10-r0" py3-pip
-RUN pip3 install --upgrade pip virtualenv
+RUN apk add --update --no-cache \
+    "python3=${PYTHON_VERSION}" \
+    py3-pip
+RUN pip3 install --upgrade \
+    pip \
+    virtualenv
 
 RUN mkdir -p /venv
 RUN python3 -m venv /venv
 
 # add python app requirements
-COPY requirements_app.txt /tmp/requirements_app.txt
-RUN /venv/bin/pip3 install -r /tmp/requirements_app.txt
+COPY requirements/base.txt /tmp/requirements.txt
+RUN /venv/bin/pip3 install -r /tmp/requirements.txt
 
 COPY static-content/hello-world.txt /www/hello-world.txt
-RUN mkdir -p /www/inputs
-RUN mkdir -p /www/outputs
 
-RUN mkdir -p /workdir
-COPY src/dy_static_file_server /workdir/dy_static_file_server
+RUN mkdir -p ${INPUT_DIR} && \
+    mkdir -p ${OUTPUT_DIR} && \
+    mkdir -p ${WORKDIR}
+
+COPY src/dy_static_file_server ${WORKDIR}/dy_static_file_server
 
 ENV NGINX_SERVER_ROOT="/www"
 
 FROM base as production
```

Comment:
minor: not sure this is very useful
since your base contains everything, your ```production``` tag contains exactly the same.